static/js/message_edit.js

```diff
@@ -81,7 +85,7 @@ export function get_editability(message, edit_limit_seconds_buffer = 0) {
     if (!message) {
         return editability_types.NO;
     }
-    if (!is_topic_editable(message, edit_limit_seconds_buffer)) {
+    if (!is_topic_editable(message, edit_limit_seconds_buffer) && message.type === "stream") {
```

Comment:
Hmm, I'm not sure that this is correct -- we may have been relying on logic in `is_topic_editable` even for PMs before this change.  Do you have an analysis for why this is safe to do?

Reply:
This is quite safe here because this checks for the editability of a message. For PMs it is either `FULL`, `NO` or `NO_LONGER`. Only conditions for `FULL` is when editing is allowed and not time out checked in line 118 and 110. `NO` is when it is not allowed and there are only a few conditions(Line 58 is checked in Line 106, Line 63 and 70 need not be checked for Private, and conditions on line 66 and &6 are checked in line 118). So things are pretty safe as if editing is not `NO` and not `FULL` either by timeout we can safely say `NO_LONGER`.